runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -73,9 +73,20 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
         fail polling_error if polling_error.is_a?(Exception)
       end
 
-      if (http_method === :put || http_method === :patch) && AsyncOperationStatus.is_successful_status(polling_state.status) && polling_state.resource.nil?
-        get_request = MsRest::HttpOperationRequest.new(request.base_uri, request.build_path.to_s, :get, {query_params: request.query_params, headers: request.headers})
-        update_state_from_get_resource_operation(get_request, polling_state, custom_deserialization_block)
+      if AsyncOperationStatus.is_successful_status(polling_state.status)
+        # Process long-running PUT/PATCH
+        if (http_method === :put || http_method === :patch) && polling_state.resource.nil?
+          get_request = MsRest::HttpOperationRequest.new(request.base_uri, request.build_path.to_s, :get, {query_params: request.query_params, headers: request.headers})
+          update_state_from_get_resource_operation(get_request, polling_state, custom_deserialization_block)
+        end
+
+        # Process long-running POST/DELETE operation with schema defined on success status codes
+        if (http_method === :post || http_method === :delete) && custom_deserialization_block && polling_state.response
```

Comment:
Sure, The issue was we were not handling long-running POST & DELETE operation when swagger has defined a schema on the response codes. That results in the SDK method returning nil values. To fix that, sarangan made a fixes in runtime, that caused breaking all other long-running operations [Reference: https://travis-ci.org/Azure/azure-sdk-for-ruby/jobs/199045822#L1149]. 

Now To fix that I've reverted his changes and made a simple change on how we handle POST/DELETE long-running operation. The one thing to note is that, we do not want to call get operation because these are POST/DELETE operation and that's why I wrapped them differently from PUT/PATCH (if condition just above this code).